src/main/java/org/zalando/nakadi/repository/db/SubscriptionDbRepository.java

```diff
@@ -84,6 +111,8 @@ public Subscription getSubscription(final String owningApplication, final Set<St
             throw new InternalNakadiException("Serialization problem during getting event type", e);
         } catch (final EmptyResultDataAccessException e) {
             throw new NoSuchSubscriptionException("Subscription does not exist", e);
+        } catch (final DataAccessException e) {
+            throw new ServiceUnavailableException("Error occurred when running database request", e);
```

Comment:
Hmm, probably you are right. We should log the DataAccessException to our LOG and not give it to user.